### PR TITLE
feat(provider/kubernetes-v2): Add pod logs output for job kinds in deploy manifest

### PR DESCRIPTION
**Pull request by benjaminws**
Not sure this is the best idea because there's a modal for this already in angular, but I couldn't figure out the path to using it in a react component. Also, note this could be expanded to other types, but I didn't see the need at this point. It would require only some mild refactoring.

job-console

log-modal

- Addresses https://github.com/spinnaker/spinnaker/issues/3029

### CODE REVIEW

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  public open() {
+    this.setState({ showModal: true });
+  }
+
+  public onClick() {
+    const { manifestEvent, accountId } = this.props;
+    // NOTE(benjaminws): This is not great, but can't get the pod name from the event data
+    const podName = `pod ${trim(manifestEvent.message.split(':')[1])}`;
```
[the remark this answers is not in the document]
Comment (benjaminws) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx]: These were undefined as far as I could tell (which seems weird).

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+// JobManifestPodLogs exposes pod logs for Job type manifests in the deploy manifest stage
+export class JobManifestPodLogs extends React.Component<IJobManifestPodLogs, IJobManifestPodLogs> {
+  constructor(props: IJobManifestPodLogs) {
+    super(props);
+    this.state = {
+      output: '',
+      showModal: false,
+      ...props,
```
[the remark this answers is not in the document]
Comment (benjaminws) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx]: Mild refactor incoming to make this make more sense.

**Comment by benjaminws** [comment]
> This is great. Ideally I'd like the link placement to match what's already present in the UI, could we include it next to the existing 'YAML' and 'Details' links in ManifestStatus.tsx and in its own CollapsibleSection in ResourceDetails.tsx to match what we see in instance details?

I am not sure this will work simply because this action (viewing console logs) is contextual to the pods themselves, not the manifest with the spec that creates them. Hope that makes sense.

> It makes the template logic slightly more verbose in those two files (since you need to check whether any event startsWith('Created pod'))[1] but would allow the links to match layout of existing console links and would break the dependency that JobManifestPodLogs has on IManifestEvent (along with the accountId change mentioned in another comment).

The reason we're coupled to IManifestEvent here is kind of on purpose. We want to be able to view the logs for the pods that are created as a result of a kubernetes operation. In this case, a job. The way the UX appears to work now is to use kubernetes events to show status of things like pod creation. This makes some sense to me, so for this I use the event data to glean the pod name, allowing me to then query the kubernetes service for logs about the pod.

Hope this clears up my intent here.

> [1] Something like:

>
```
const showLogs = get(manifest, 'events', []).some(e => e.message.startsWith('Created pod'));
// ...
{ showLogs && <JobManifestPodLogs manifest={manifest} linkName="Console Output (Raw)" /> }
```

**Comment by sbwsg** [comment]
Aaaah I think I understand now.  A k8s Job has a one-to-many relationship with Pods.  We'd like to be able to access log output for each pod that is run during a Job's execution.

**Comment by benjaminws** [comment]
> A k8s Job has a one-to-many relationship with Pods.

You got it.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    if (this.canShow()) {
+      return (
+        <div>
+          <a onClick={this.onClick} className="clickable">
+            {this.props.linkName}
+          </a>
+          <Modal show={showModal} onHide={this.close}>
+            <Modal.Header closeButton={true}>
+              <Modal.Title>Console Output</Modal.Title>
```
[the remark this answers is not in the document]
Comment (benjaminws) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx]: Yeah, that's probably worthwhile. I thought about it but didn't take the time. Easy enough to do.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+// IJobManifestPodLogs is the data needed to get logs
+export interface IJobManifestPodLogsProps {
+  manifest: IManifest;
+  manifestEvent: IManifestEvent;
+  linkName: string;
+}
+
+export interface IJobManifestPodLogsState {
+  showModal?: boolean;
+  output?: string;
```
[the remark this answers is not in the document]
Comment (benjaminws) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/JobManifestPodLogs.tsx]: Agreed. Fixing.

**Pull request merged**